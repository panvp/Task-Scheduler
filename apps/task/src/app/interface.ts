export interface taskListType{
    title: string;
    desciption: string;
    dropdownValue: string;
}

export interface ViewAllTaskPropsType{
    allTaskList: Array<taskListType>;
    deleteTaskHandler: (index: number) => void;
    editTaskHandler: (index: number) => void;
    filterValue: string;
}
export interface newTaskPropsType{
    isEdit: boolean;
    setIsEdit: React.Dispatch<React.SetStateAction<boolean>>;
    indexToEdit: number;
    taskToBeEdited: taskListType; 
    setAllTaskList: React.Dispatch<React.SetStateAction<Array<taskListType>>>;
}

