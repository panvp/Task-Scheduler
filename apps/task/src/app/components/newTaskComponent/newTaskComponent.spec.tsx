import { render, screen } from '@testing-library/react';
import { taskListType } from '../../interface';
import NewTaskComponent from './newTaskComponent.component';

describe('Home Screen Test Cases', () => {
  beforeEach(() => {
    
    const isEdit = false;
    const setIsEdit = jest.fn();
    const setAllTaskList = jest.fn();
    const indexToEdit = 1;
    const taskToBeEdited = {title: "", description: "", dropdownValue: ""}
    render(
      <NewTaskComponent
        isEdit={isEdit}
        setIsEdit={setIsEdit}
        indexToEdit={indexToEdit}
        taskToBeEdited={taskToBeEdited}
        setAllTaskList={setAllTaskList}
      />
    );
  });
  test('Snapshot testing', () => {
    const homeScreen = screen.getByTestId('newTaskComponent');
    expect(homeScreen).toMatchSnapshot();
  });
});
