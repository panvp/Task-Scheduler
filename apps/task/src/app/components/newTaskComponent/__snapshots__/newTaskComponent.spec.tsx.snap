// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Test Cases Snapshot testing 1`] = `
<dialog
  class="modal-newTask"
  data-testid="newTaskComponent"
>
  <div
    class="m-4 p-4 md:mt-4 md:mb-8 md:p-8 flex flex-col justify-around items-start"
  >
    <h1
      class="text-2xl font-bold self-center"
    >
       
      Add your task
    </h1>
    <div
      class="flex flex-wrap mt-2 md:mt-4"
    >
      <div>
        <label
          for="titleInput"
        >
           Task Name: 
        </label>
        <input
          class="sm:mx-2"
          id="titleInput"
          placeholder="Enter name here"
          size="12"
          type="text"
          value=""
        />
      </div>
      <select>
        <option>
          To Do
        </option>
        <option>
          In Progress
        </option>
        <option>
          Done
        </option>
      </select>
    </div>
    <div
      class="mt-2 md:mt-4"
    >
      <label
        class=""
        for="titleInput"
      >
         Task Description: 
      </label>
      <input
        class="sm:mx-2"
        id="titleInput"
        placeholder="Enter here"
        type="text"
        value=""
      />
    </div>
    <button
      class="font-medium rounded-lg text-sm px-2 md:px-8 py-2 mt-2 md:mt-4 text-center self-center bg-slate-200"
      disabled=""
    >
       
      Add task
    </button>
  </div>
</dialog>
`;
